### PR TITLE
Upgrade Downshift

### PR DESCRIPTION
**Pull request by Rowno**
This upgrades Downshift from v1 to v3 and fixes a couple of bugs.

### Bug fixes
- Pressing the up arrow key in the text input now correctly takes you to the bottom of the list.
- The Popover's Portal is now properly supported which improves accessibility and fixes touch screen support.
- Anything using Positioner will now render correctly if `isShown` is `true` on initial render.

### Breaking changes
Because all the Downshift props are exposed directly, these breaking changes are introduced to `Autocomplete` and `Combobox`:
- https://github.com/paypal/downshift/releases/tag/v2.0.0
- https://github.com/paypal/downshift/releases/tag/v3.0.0

### CODE REVIEW

File: src/autocomplete/src/Autocomplete.js [hunk 1]
```diff
@@ -163,10 +168,7 @@ export default class Autocomplete extends PureComponent {
                   index,
                   style,
                   children: itemString,
-                  onMouseUp: () => {
-                    selectItemAtIndex(index)
-                  },
-                  isSelected: itemToString(selectedItem) === itemString,
+                  isSelected: selectedItem === item,
```
Comment (Rowno) [inline src/autocomplete/src/Autocomplete.js]: This avoids an unnecessary string conversion.

File: src/table/src/TableRow.js [hunk 2]
```diff
@@ -144,8 +144,8 @@ class TableRow extends PureComponent {
           innerRef={this.onRef}
           className={cx(themedClassName, className)}
           display="flex"
-          aria-selected={isHighlighted}
-          aria-current={isSelected}
+          data-ishighlighted={isHighlighted}
+          aria-selected={isSelected}
```
Comment (Rowno) [inline src/table/src/TableRow.js]: Downshift adds `aria-selected` and it behaves like `isSelected`. Using both `aria-current` and `aria-selected` also probably isn't correct accessibility wise so I switched `isHighlighted` to a data attribute. I tested in a screen reader and highlights are still being announced correctly by Downshift.

Comment (jeroenransijn) [reply src/table/src/TableRow.js]: Yeah those `aria-` attributes keep tripping me up. We might need to check if this works for all the consumers of `TableRow`.

File: src/autocomplete/src/Autocomplete.js (hunk 1 above)
Comment (Rowno) [reply src/autocomplete/src/Autocomplete.js]: This is how they do in the Downshift docs though? https://github.com/paypal/downshift#usage

Comment (Rowno) [reply src/autocomplete/src/Autocomplete.js]: I think you're right, I'll revert this change.

File: src/table/src/TableRow.js (hunk 2 above)
Comment (Rowno) [reply src/table/src/TableRow.js]: I just tested `Menu`, which from what I can tell is the only other consumer of `createRowAppearance` (nothing else uses `TableRow`), and the highlighted / selected states are still working there.

Comment (Rowno) [reply src/table/src/TableRow.js]: `Table` itself also seems to be working fine.

**Pull request merged**